### DISCUSSION TITLE
How to add custom configuration / attributes to an client

### DISCUSSION THREAD

**Discussion by FlxPeters** [keycloak/keycloak#30153] [Q&A]
I have a question regarding clients and some kind of metadata about these clients:

I try to implement some conditional authentication steps in my authentication flow based on client attributes.

For example, I want to mark all business critical clients in order to perform further checks on the IP, session age and so on.
This could be implemented via a custom condition in the authentication flow and custom authenticators triggered on these conditions.

The problem is:  There is no option to add this information to my clients.
At least no option which is fully supported via the UI.
Clients have an `attributes` map in the API, but it's not displayed in the UI. 

So how do you solve such an issue? How to add custom attributes to a client in order to annotate it for further processing?

I assumed the following options:

* **Use the Client attributes** - Not sure if it's a good idea. Is there a reason why they are not accessible via the UI?
* **Add a dummy client-role** - A role could be misused as key-value store since it has a name and an attribute store. Feels some kind of dirty, but is fully supported via API and UI.
* Any other ideas? 

Looking forward to your feedback. I'm sure I'm not the only one with this issue, right?

**Comment by codespearhead** [comment]
I suggest closing this discussion and promote the reopening of #6250 by voicing your use case there.

Possible duplicates: #9626 .

**Reply by FlxPeters** [reply]
Thanks for the feedback. I have already seen this PR, but it seems there is no traction for 4 years now.

I think I will use the attributes and start a new PR for client attributes in the account console.
In the React based version, this should be easy.

**Reply by achetronic** [reply]
I would like to see the attributes through the UI. Just for checking purposes at least
